src/com/tigersale/userInterface/customerUser/ListOfProductView.java

```diff
@@ -0,0 +1,195 @@
+package com.tigersale.userInterface.customerUser;
+
+import com.tigersale.database.ProductTable;
+import com.tigersale.database.ShoppingListTable;
+import com.tigersale.model.CustomerUser;
+import com.tigersale.model.Product;
+import com.tigersale.userInterface.AbstractView;
+
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * List of Products View
+ * @author My Tran
+ */
+public class ListOfProductView extends AbstractView {
+
+    /**
+     * The logged in user
+     */
+    private CustomerUser user;
+    /**
+     * Constructor for the ListOfProductView
+     * @param scanner The scanner to retrieve user input from
+     */
+    public ListOfProductView(Scanner scanner, CustomerUser user) {
+        super(scanner);
+        this.user = user;
+    }
+
+    /**
+     * This method acts as the view for the CustomerUser to view the Product list
+     */
+    public void runListOfProductView() {
+        while (true) {
+            System.out.println("=========================================================");
+            System.out.println("                     List of Products                    ");
+            System.out.println("=========================================================");
+
+            //Retrieve the list of products from the database
+            List<Product> products = ProductTable.viewProducts();
+            int size = products.size();
+            int choice;
+
+            if (size == 0)
+            {
+                System.out.println("There is no products in the system yet");
+                return;
+            } else
+            {
+                System.out.println("Please type in an integer corresponding to your preferred option.");
+                System.out.println(0 + ": Go Back");
+                System.out.println(1 + ": View Shopping List");
+                System.out.println("OR");
+                System.out.println("-----------------------------------------------------");
+                //Print all the products with associated numbers
+                System.out.println("Please choose the number associated with the product to view detail:");
+                int i = 0;
+                while (i < size) {
+                    System.out.println((i+2) + ": " + products.get(i).name);
+                    i++;
+                }
+
+
+                // Try to get a numeric response from the user
+                try {
+                    choice = scanner.nextInt();
+                } catch (InputMismatchException e) {
+                    System.out.println("Please type in an integer corresponding to your preferred option.");
+                    scanner.next();
+                    continue;
+                }
+
+                //Invalid option
+                if (choice > (size + 1))
```

Comment:
Use switch statement here. You can do
switch(choice)
{
case 0:
...
break;
case 1:
...
break;
default:
// this is your error case or "choice > 1"
break;